### DISCUSSION TITLE
Seperate data location of database from index

### DISCUSSION THREAD

**Discussion by CyberCyclone** [ravendb/ravendb#14327] [General]
Hi everyone,
I'm looking at shifting the main DB data onto a slower SATA based SSD due to the fact the size of the DB is pretty big and it's getting expensive. I want to keep just the index data on the faster disk.

Is it possible to set the location of Indexes so they're on the faster NVME drives while the main database is stored on a slower SATA SSD? When creating the index I couldn't see an option to set the index disk location and I haven't seen any configuration I can set.

From reading this doc: https://ravendb.net/docs/article-page/5.3/csharp/server/storage/customizing-raven-data-files-locations#linux-1, it seems to suggest that I can create a soft link to the location via Linux system: `ln -s ~/RavenDB/Server/RavenData/Databases/Northwind/Indexes /mnt/FastDrive/Databases/Northwind/Indexes`

Is this the best approach? I'd rather keep the config inside RavenDB otherwise it's a manually process.

Thanks!
Casey.

**Comment by ayende** [comment]
Yes, soft links are the right way to handle that.

What is the total size of the database and its indexes? What kind of queries are you usually running?

**Reply by CyberCyclone** [reply]
Awesome, thanks. 

The database is about 250GB of extracted blockchain data. So it has very minimum writes and mostly reads. Each index is about 2 - 10GB in size.

**Reply by CyberCyclone** [reply]
Could it be put in the roadmap to add an option to set an index location when it's being created like when a DB is created?

**Reply by ayende** [reply]
No, that is quite explicitly something that we don't want to do.
The reason is that adding that as a configuration creates a whole *messy* situation.

On the other hand, you can define that as a single location using: `Storage.OnDirectoryInitialize.Exec`

See: https://ravendb.net/docs/article-page/5.3/csharp/server/storage/customizing-raven-data-files-locations#automation

**Reply by CyberCyclone** [reply]
Thanks for that, I did see that and I was just making sure that it was the best approach. 

Thanks for your help! It looks like it will definitely solve my issue.
